### PR TITLE
Feature/question accessible error handling

### LINKED ISSUE #91: 2.1: Kyselyelementtien virheidenkäsittelyn ongelmia
ks. Liite 1 Saavutettavuustestausraportti luku 4.6 (kuvaus ja vaikutus, korjausehdotus, WCAG-onnistumiskriteerit).

### PR DESCRIPTION
**Pull request by mmoila**
Major changes:
- It's now only possible to navigate between survey pages using next and previous buttons. 
- New page specific error message is displayed if a user tries to continue to next page or submit the survey without filling required questions. 

Closes #91. See #91 for changelog.

### CODE REVIEW

File: client/src/components/CheckBoxQuestion.tsx [hunk 1]
```diff
@@ -58,12 +61,19 @@ export default function CheckBoxQuestion({
...
+          {validationErrors && validationErrors.includes('answerLimits') && (
+            <FormHelperText style={visuallyHidden} role="alert">
+              {`${question.title?.[surveyLanguage]}, ${answerLimitText}`}
+            </FormHelperText>
+          )}
+        </>
       )}
       <FormGroup
-        id={`${question.id}-input`}
+        aria-describedby="checkbox-helper-label"
```
Comment (jpeltzi) [inline client/src/components/CheckBoxQuestion.tsx:76]: Aria description here is obsolete, as the `<FormGroup>` rendering as mere `<div>` will not be able to receive focus.
Also, won't there be an ID clash if there are multiple checkbox questions per page?

**Review by jpeltzi — CHANGES_REQUESTED** [review]

File: client/src/components/SurveyQuestion.tsx [hunk 2]
```diff
@@ -21,46 +22,78 @@ import SortingQuestion from './SortingQuestion';
...
+          color: survey.sectionTitleColor ?? '#000000',
         }}
       >
-        <FormLabel
-          htmlFor={`${question.id}-input`}
-          style={{ color: survey.sectionTitleColor ?? '#000000' }}
-        >
-          {question.title?.[surveyLanguage]} {question.isRequired && '*'}
-        </FormLabel>
+        <h3 tabIndex={0}>
```
[the remark this answers is not in the document]
Comment (mmoila) [reply client/src/components/SurveyQuestion.tsx:84]: `tabindex`removed

File: client/src/components/SurveyStepper.tsx [hunk 3]
```diff
@@ -289,102 +305,135 @@ export default function SurveyStepper({
...
+                </h2>
+              </StepLabel>
+
+              <StepContent
+                transitionDuration={0}
+                classes={{ root: classes.stepContent }}
+              >
+                {pageUnfinished && (
+                  <Alert
+                    tabIndex={0}
```
[the remark this answers is not in the document]
Comment (mmoila) [reply client/src/components/SurveyStepper.tsx:338]: Fixed as suggested

File: client/src/components/SurveyStepper.tsx [hunk 4]
```diff
@@ -289,102 +305,135 @@ export default function SurveyStepper({
...
+                    ref={currentPageErrorRef}
+                    severity="error"
```
[the remark this answers is not in the document]
Comment (mmoila) [reply client/src/components/SurveyStepper.tsx:340]: Fixed as suggested

File: client/src/components/CheckBoxQuestion.tsx (hunk 1 above)
Comment (mmoila) [reply client/src/components/CheckBoxQuestion.tsx:76]: Removed `aria-describedby` and fixed id for possible future use

**Pull request merged**